tools/upgrade.sh

```diff
@@ -138,39 +148,30 @@ check_for_updates() {
         perform_upgrade $latest_release
     else
         printf "${BOLD}${GREEN}✅ No updates needed!${RESET} \"spin\" is up-to-date. Now get back to work! \n"
-        echo $(date +"%s") > $SPIN_HOME/conf/last_update_check.lock
+        date "+%s" > "${SPIN_HOME}/conf/last_update_check.lock"
     fi
 }
 
 get_current_version() {
-    local local_tag
-    local_tag=$(git -C $SPIN_HOME describe --tags --abbrev=0)
-
-    echo $local_tag
-
+    git -C "$SPIN_HOME" describe --tags --abbrev=0
 }
 
 get_latest_release() {
+    source "$SPIN_CONFIG_FILE_LOCATION"
 
-    source $SPIN_CONFIG_FILE_LOCATION
-
-    if [ "$TRACK" == "beta" ]; then
+    if [ "$TRACK" == beta ]; then
         # Get the latest release (including pre-releases). We just want the 
         # absolute latest release, regardless of pre-release or stable
-        curl --silent \
-            -H "Accept: application/vnd.github.v3+json" \
-            "https://api.github.com/repos/serversideup/spin/releases" | \
-        grep '"tag_name":' | \
-        sed -E 's/.*"([^"]+)".*/\1/' | \
-        head -n 1
-    else
-        # Get latest stable release
-        curl --silent \
-            -H 'Accept: application/vnd.github.v3.sha' \
-            "https://api.github.com/repos/serversideup/spin/releases/latest" | \
-        grep '"tag_name":' | \
-        sed -E 's/.*"([^"]+)".*/\1/'
+        curl --silent --header "Accept: application/vnd.github.v3+json" \
+            "https://api.github.com/repos/serversideup/spin/releases" \
+            | jq -r '.[0]."tag_name"'
+        return
```

Comment:
Here comes `jq` 🚀

Reply:
`jq` is great, but I do not want to have any dependencies. (`jq` does not ship with Mac by default, that's why I used `sed`).

Thoughts?

Reply:
...meaning I could not produce a tool/package for non-working people who do not have essential tools.

But what does essential mean ❓

Reply:
@jaydrogers Does Docker ship with Mac? 😲

Reply:
LOL, other than installing Docker Desktop.... My goal is to make it as easy as possible to get started with little dependencies as possible.

I already went through the pain of getting `sed` to work, so I don't see any other value to switching it to `jq` for the time being.

Reply:
> LOL, other than installing Docker Desktop....

Do you mention installing it in the docs?

Reply:
LOL, whoops 😅

It does now. Thanks for catching that! https://serversideup.net/open-source/spin/installation/install-macos